### PR TITLE
MUI6 upgrade

### PR DESCRIPTION
**Pull request by shubham-eaton**
#### Changes proposed in this Pull Request:

- Updated MUI version to 6
- Updated demo showcase to use MUI6
-

#### Screenshots / Screen Recording (if applicable)

-
Screenshot 2024-10-28 at 1 21 54 PM

#### To Test:

- cd demo/showcase
- yarn add @mui/codemod @mui/docs @mui/envinfo @mui/icons-material @mui/lab @mui/material @mui/private-theming  @mui/styled-engine @mui/styled-engine-sc @mui/styles @mui/system @mui/types @mui/x-date-pickers
- yarn start
- Open RTLProvider.tsx 
- import -> import {blueThemes} from '@brightlayer-ui/react-themes'
- Replace theme provider -> `<ThemeProvider
        theme={blueThemes}
        defaultMode='light'
        >`

### CODE REVIEW

**Comment by JeffGreiner-eaton** [comment]
I see in PR you added showcase demo screen shot. (do you have a working version of demo? I had to make adjustments)
We should have some test instructions in here for using react-showcase-demo branch [feature/mui-6](https://github.com/etn-ccis/blui-react-showcase-demo/compare/dev...feature/mui-6) if we are trying to see what's working and not working...

- do we know why @mui/styled-engine version 6 is throwing error?
No matching exports on internal_serializeStyles and internal_mutateStyles

**Comment by shubham-eaton** [comment]
> I see in PR you added showcase demo screen shot. (do you have a working version of demo? I had to make adjustments) We should have some test instructions in here for using react-showcase-demo branch [feature/mui-6](https://github.com/etn-ccis/blui-react-showcase-demo/compare/dev...feature/mui-6) if we are trying to see what's working and not working...
> 
> * do we know why @mui/styled-engine version 6 is throwing error?
>   No matching exports on internal_serializeStyles and internal_mutateStyles
> * are all these new component styles overrides the new implementation and does this line us up for mui-v7?

Hi @JeffGreiner-eaton  , I have updated the description for testing MUI 6 changes. 
For  @mui/styled-engine I am not getting any error. Maybe we can connect offline to check this.
MUI-v7 is mainly having changes in few of their components, hence these styles overrides and theme related changes should work fine with MUI-v7 also.

File: scripts/buildTest.sh [hunk 1]
```diff
@@ -16,10 +16,8 @@ if [ ! -f ./LICENSES.json ]; then echo -e "${RED}Not Found${NC}" && exit 1; else
 echo -ne "  changelog: "
 if [ ! -f ./CHANGELOG.md ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
 
-echo -ne "  blue theme: "
-if [ ! -f ./dist/blueTheme.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
-echo -ne "  blue dark theme: "
-if [ ! -f ./dist/blueDarkTheme.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
+echo -ne "  blue merged theme: "
+if [ ! -f ./dist/blueMergedTheme.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
```
Comment (ektaghag-eaton) [inline scripts/buildTest.sh:20]: are we going to use this name or `blueTheme.js` ?

Comment (ektaghag-eaton) [inline scripts/buildTest.sh:20]: are we going to use this name or `blueTheme.js` ?

Comment (shubham-eaton) [reply scripts/buildTest.sh:20]: We are going to use the name blueThemes

Comment (shubham-eaton) [reply scripts/buildTest.sh:20]: We are going to use the name blueThemes

**Comment by JeffGreiner-eaton** [comment]
I updated per instructions, but I may of missed something for RTL. Is RTL working ok for you?

image

**Comment by shubham-eaton** [comment]
> I updated per instructions, but I may of missed something for RTL. Is RTL working ok for you?
> 
> image

It is same for me as well

**Review by JeffGreiner-eaton — APPROVED** [review]
did side-by-side comparison to dev branch, looks good

**Pull request merged**